benchmark.go

```diff
@@ -39,8 +42,102 @@ import (
 var (
 	errNegativeDuration = errors.New("duration cannot be negative")
 	errNegativeMaxReqs  = errors.New("max requests cannot be negative")
+
+	// using a global _quantiles slice and _percentileMap mainly for ease of testing, and not passing the same array
+	// around to multiple functions
```

Comment:
while passing around the globals may be easier, we likely will want to pass in the quantiles as a flag in future.

Although that makes me wonder, do we really want the hardcoded mapping from quantile to a name like P999? Can we either:
 - generate them
 - or just keep it simple and use the percentile value (perhaps with a fixed number of decimal places as the key)

cc @jaricftw